### PR TITLE
O auth2

### PR DESCRIPTION
**Pull request by Shivankshikh**
Hi..
I have enabled the OAUth2 using google single sign on for the web application security. Please review the code.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -52,6 +52,44 @@
 			</exclusions>
 		</dependency>
 		
+		<!-- to use JSP as the view. -->
+		<dependency>
+		    <groupId>org.apache.tomcat.embed</groupId>
+		    <artifactId>tomcat-embed-jasper</artifactId>
```
Comment (abhijay-kumar) [inline pom.xml:58]: Why do you need this ?

File: src/main/java/com/shivankshi/emscrud/EmscrudApplication.java [hunk 2]
```diff
@@ -1,13 +1,26 @@
...
+
+
 public class EmscrudApplication {
 
 	public static void main(String[] args) {
 		SpringApplication.run(EmscrudApplication.class, args);
 	}
 
+	
+//	@Bean
```
Comment (abhijay-kumar) [inline src/main/java/com/shivankshi/emscrud/EmscrudApplication.java:18]: Remove commented code

File: src/main/java/com/shivankshi/emscrud/config/SwaggerConfig.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import springfox.documentation.spi.service.contexts.SecurityContext;
+import springfox.documentation.spring.web.plugins.Docket;
+import springfox.documentation.swagger.web.SecurityConfiguration;
+import springfox.documentation.swagger.web.SecurityConfigurationBuilder;
+import springfox.documentation.swagger2.annotations.EnableSwagger2;
+
+@Configuration
+@EnableSwagger2
+
+public class SwaggerConfig {
```
Comment (abhijay-kumar) [inline src/main/java/com/shivankshi/emscrud/config/SwaggerConfig.java:27]: This sould be in another PR.

File: src/main/java/com/shivankshi/emscrud/rest/HomeController.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+@RestController
+public class HomeController {
+	
+	Logger logger=LoggerFactory.getLogger(HomeController.class);
+	
+	@RequestMapping("/")
+	public String home()
+	{
+		logger.info("welcome in the EMS");
```
Comment (abhijay-kumar) [inline src/main/java/com/shivankshi/emscrud/rest/HomeController.java:16]: Turn on formatter.

File: src/main/java/com/shivankshi/emscrud/service/EmployeeServiceImpl.java [hunk 5]
```diff
@@ -19,7 +19,7 @@
 	
 	@Autowired
 	public EmployeeServiceImpl(EmployeeRepository employeeRepository) {
-		//bean id due to conflict in DAO implementation
```
Comment (abhijay-kumar) [inline src/main/java/com/shivankshi/emscrud/service/EmployeeServiceImpl.java:22]: Try to use javadoc comments while describing anything

File: src/main/java/com/shivankshi/emscrud/service/EmployeeServiceImpl.java [hunk 6]
```diff
@@ -48,11 +48,6 @@ public Employee findById(int theId) {
 		return employee;
 	}
 
-//	@Override
```
Comment (abhijay-kumar) [inline src/main/java/com/shivankshi/emscrud/service/EmployeeServiceImpl.java:51]: Remove commented code.

File: src/main/java/com/shivankshi/emscrud/service/EmployeeServiceImpl.java [hunk 7]
```diff
@@ -61,29 +56,29 @@ public void deleteById(int theId) {
...
+		Optional<Employee> employeeFromDb=employeeRepository.findById(empId);
+		Employee employee2=null;
+		if(employeeFromDb.isPresent())
+		{
+			employee2=employeeFromDb.get();
+			employee2.setDesignation(employee.getDesignation());
+			employee2.setSalary(employee.getSalary());
+		}
+		else {
+			throw new RuntimeException("Employee id not found");
```
Comment (abhijay-kumar) [inline src/main/java/com/shivankshi/emscrud/service/EmployeeServiceImpl.java:78]: Log your exceptions

File: src/test/java/com/shivankshi/emscrud/rest/EmployeeRestControllerTest.java [hunk 8]
```diff
@@ -121,18 +158,19 @@ public void testAddEmployee() throws Exception
...
 		
-		MvcResult result=mockMvc.perform(requestBuilder).andReturn();
+		MvcResult result=mockMvc.perform(requestBuilder).andDo(MockMvcResultHandlers.print())
+		        .andExpect(status().isOk()).andReturn();
 		
 		MockHttpServletResponse response=result.getResponse();
 		
 		String outputInJsonString=response.getContentAsString();
-		
+		System.out.println(outputInJsonString);
```
Comment (abhijay-kumar) [inline src/test/java/com/shivankshi/emscrud/rest/EmployeeRestControllerTest.java:167]: Do not use sysout, always use loggers.

Comment (Shivankshikh) [reply src/test/java/com/shivankshi/emscrud/rest/EmployeeRestControllerTest.java:167]: okay.
